src/Razor/src/Microsoft.AspNetCore.Razor.Language/Components/ComponentSplatLoweringPass.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Runtime.InteropServices.ComTypes;
+using Microsoft.AspNetCore.Razor.Language.Intermediate;
+
+namespace Microsoft.AspNetCore.Razor.Language.Components
+{
+    internal class ComponentSplatLoweringPass : ComponentIntermediateNodePassBase, IRazorOptimizationPass
+    {
+        // Run after component lowering pass
+        public override int Order => 50;
+
+        protected override void ExecuteCore(RazorCodeDocument codeDocument, DocumentIntermediateNode documentNode)
+        {
+            if (!IsComponentDocument(documentNode))
+            {
+                return;
+            }
+
+            var references = documentNode.FindDescendantReferences<TagHelperPropertyIntermediateNode>();
```

Comment:
Should rebase the PR onto https://github.com/aspnet/AspNetCore-Tooling/pull/638 so you can use the directive attribute intermediate node goods. You'll have to anyways otherwise all this wont work as intended once both PRs are merged.

Reply:
https://github.com/aspnet/AspNetCore-Tooling/pull/638 is merged. @rynowak you'll need to rebase and change some of this logic.

Reply:
What will need to change? `@attributes` doesn't have any parameters.